cpusupport/Build/cpusupport-X86-SHANI.c

```diff
@@ -0,0 +1,17 @@
+#include <stdint.h>
+#include <immintrin.h>
+
+static uint8_t block[16] = {0};
+
+int
+main(void)
+{
+	uint32_t state[8] = {0};
```

Comment:
s/8/16/ ?

Reply:
Hmm, don't you mean 4?  It's `unit32_t`, and getting loaded into `__m128i`.

Reply:
We have
```
static void
SHA256_Transform(uint32_t state[static restrict 8],
    const uint8_t block[static restrict 64],
    uint32_t W[static restrict 64], uint32_t S[static restrict 8])
{
```

because the state is split up into two separate `__m128i` variables.  But when checking for the available of the instructions, we don't need to bother with that.

Reply:
Err, yeah.  I was confused because I think of XMM registers as being 16 bytes, not 4 doublewords.